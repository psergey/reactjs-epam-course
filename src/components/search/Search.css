.header-container {
    height: 396px;
    background-image: url("/src/assets/header.jpg");
}

.header {
    height: 100%;
    -webkit-backdrop-filter: blur(5px);
    backdrop-filter: blur(5px);
}

.flex-row {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
}

.flex-row :last-child {
    margin-left: auto;
}

.flex-column {
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    place-content: center;    
}

.row > div, .flex-column > div {
    display: flex;
}

.--header::after {
    content: '';
    position: absolute;
    height: 100%;
    width: 100%;
    top: 0;
    left: 0;
    background-image: url("/src/assets/header.jpg");
    -webkit-filter: blur(5px);
    filter: blur(5px);
    z-index: -1;
}

.header .roulette-label {
    text-align: left; 
}

.header .flex-row {
    padding: 20px 0 57px 60px;
}

input.search {
    height: 57px;
    width: 713px;
    background: rgba(50, 50, 50, 0.8);
    border: 0px;
    border-radius: 4px;    
    margin: 0 14px 0 0;
    padding: 0 0 0 19px;
    color: #ffffff;
    font-size: 20px;
    line-height: 24px;
}

button.search {
    background: #F65261;
    height: 57px;
    min-width: 233px;
    text-transform: capitalize;
    border: 0px;
    border-radius: 4px;
    color: #ffffff;
    font-size: 20px;
    line-height: 24px;
    text-align: center;
    text-transform: uppercase;
}

span.search {
    font-size: 40px;
    line-height: 49px;
    font-weight: 300;
    text-align: center;
    letter-spacing: 1px;
    text-transform: uppercase;
    color: #FFFFFF;    
}

button.addMovie {
    font-weight: 600;
    font-size: 20px;
    line-height: 24px;
    color: #F65261;
    padding: 11px 18px;
    width: 177px;
    border: 0;
    border-radius: 4px;
    height: 46px;
    background: rgba(96, 96, 96, 0.68);    
    text-align: right;    
    text-transform: uppercase;
    margin-right: 50px;
}