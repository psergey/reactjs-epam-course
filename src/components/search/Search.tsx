import { FC, ReactElement } from "react";
import RouletteLabel from "../roulette-label/RouletteLabel";
import "./Search.css"

const Search : FC = () : ReactElement => {
    return (
        <div className="header-container">
            <div className="header">
                <div className="flex-row">
                    <RouletteLabel />
                    <button className="addMovie">
                        + Add Movie
                    </button>
                </div>
                <div className="flex-column">
                    <div><span className="search">Find your movie</span></div>
                    <div>
                        <div><input placeholder="What do you want to watch?" className="search"></input></div>
                        <div><button className="search">Search</button></div>
                    </div>
                </div>
            </div>
        </div>
    )
}

export default Search;