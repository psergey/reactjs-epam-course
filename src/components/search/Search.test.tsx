import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Search from './Search';

describe('search component tests', () => {
  it('should match to snaphot', () => {
    const query = 'Doctor Who';
    const searchFn = jest.fn();
    const { asFragment } = render(<Search query={query} onSearch={() => searchFn} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render component with provided search qriteria', () => {
    const query = 'Doctor Who';
    const searchFn = jest.fn();
    const { getByRole } = render(<Search query={query} onSearch={() => searchFn} />);

    const input = getByRole('textbox');

    expect(input).toHaveValue(query);
  });

  it('should pass search criteria by clicking on search button', async () => {
    const query = 'Doctor Who';
    const searchFn = jest.fn();
    const { getByRole } = render(<Search onSearch={(searchCriteria: string): void => searchFn(searchCriteria)} />);

    const input = getByRole('textbox');
    const btn = getByRole('button');

    await userEvent.type(input, query);
    await userEvent.click(btn);

    expect(searchFn).toHaveBeenCalledWith(query);
  });

  it('should pass search criteria by pressing enter on search textbox', async () => {
    const query = 'Doctor Who';
    const searchFn = jest.fn();
    const { getByRole } = render(<Search onSearch={(searchCriteria: string): void => searchFn(searchCriteria)} />);

    const input = getByRole('textbox');

    await userEvent.type(input, query);
    input.focus();
    await userEvent.keyboard('{enter}');

    expect(searchFn).toHaveBeenCalledWith(query);
  });
});
