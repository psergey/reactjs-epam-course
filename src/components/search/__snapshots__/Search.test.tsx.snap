// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search component tests should match to snaphot 1`] = `
<DocumentFragment>
  <input
    class="search"
    placeholder="What do you want to watch?"
    value="Doctor Who"
  />
  <button
    class="search"
  >
    Search
  </button>
</DocumentFragment>
`;
